hnn_core/network_builder.py

```diff
@@ -318,6 +318,8 @@ def _build(self):
         self._create_cells_and_drives(threshold=self.net._params['threshold'],
                                       record_vsoma=record_vsoma,
                                       record_isoma=record_isoma)
+        n_drive_cells = len(self._drive_cells)
+        print(f'n_art_cells = {n_drive_cells}')
```

Comment:
I'm temporarily including these lines to demo the issue. Previously, a simulation using `MPIBackend` with 6 processes would end up creating at total of 80 drive cells when 20 were expected.

Reply:
can you think of a way to test the gid assignments? I think you might have to be a little clever how you set it up. E.g., have a parameter called `rank` or `test_mode` in one of the functions that fakes the `MPIBackend` with `rank > 1`. See how we do something similar in MNE:

https://github.com/mne-tools/mne-python/blob/9b43b2beeca84931049bdc83717aa579a747f2ea/mne/report.py#L94-L97

or maybe have `rank` as an attribute that is set once somewhere and you can manually tweak it

Reply:
That's a good idea. I'll give it a shot, though I probably won't be able to get around to it until later next week.